core-api/src/main/java/com/optimizely/ab/OptimizelyUserContext.java

```diff
@@ -0,0 +1,186 @@
+/**
+ *
+ *    Copyright 2020, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab;
+
+import com.optimizely.ab.optimizelydecision.OptimizelyDecideOption;
+import com.optimizely.ab.optimizelydecision.OptimizelyDecision;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class OptimizelyUserContext {
+    @Nonnull
+    private final String userId;
+
+    @Nonnull
+    private final Map<String, Object> attributes;
+
+    @Nonnull
+    private final Optimizely optimizely;
+
+    private static final Logger logger = LoggerFactory.getLogger(OptimizelyUserContext.class);
+
+    public OptimizelyUserContext(@Nonnull Optimizely optimizely,
+                                 @Nonnull String userId,
+                                 @Nonnull Map<String, ?> attributes) {
+        this.optimizely = optimizely;
+        this.userId = userId;
+        this.attributes = new ConcurrentHashMap<>(attributes);
+    }
+
+    public OptimizelyUserContext(@Nonnull Optimizely optimizely, @Nonnull String userId) {
+        this(optimizely, userId, new HashMap<>());
+    }
+
+    public String getUserId() {
+        return userId;
+    }
+
+    public Map<String, Object> getAttributes() {
+        return new HashMap<String, Object>(attributes);
+    }
+
+    public Optimizely getOptimizely() {
+        return optimizely;
+    }
+
+    /**
+     * Set an attribute for a given key.
+     *
+     * @param key An attribute key
+     * @param value An attribute value
+     */
+    public void setAttribute(@Nonnull String key, @Nonnull Object value) {
```

Comment:
we may pass `null`

Reply:
I guess it technically won't hurt anything, but it won't be useful in any way to have a null key

Reply:
am asking for `Object` value, it can be `null`.